### PR TITLE
Allow theming of actions

### LINKED ISSUE #6773: Change fileactions icons on icons in theme folder
Steps to reproduce:
1. Put new icons in /themes/new-theme/core/img/actions/
2. Open page with filelist in browser
3. Move mouse to file

Expected results:
App uses new icons from theme folder

Actual results:
App uses old icons from /core/img/actions/ folder

I see in /core/js/js.js filePath function, and this function only uses OC.appswebroots[app] prefix and not uses theme prefix for image files.

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #6773 eventually ;)

@owncloud/designers I need your help to do this properly. We will move from calculating the path in javascript to using the icons from the icon.css file. So that styling can be done properly.

However, I need your css FU to make it look nice.

### CODE REVIEW

**Comment by rullzer** [comment]
Currently I'm styling the file actions which look like: 
actions

I replaced some of the icons with the triangle so I know stuff is different.

@owncloud/designers could you take a look at the css there to make it all work again?

File: apps/files/js/fileactions.js [hunk 1]
```diff
@@ -11,7 +11,7 @@
 (function() {
 
 	var TEMPLATE_FILE_ACTION_TRIGGER =
-		'<a class="action action-{{nameLowerCase}}" href="#" data-action="{{name}}">' +
+		'<a class="action action-{{nameLowerCase}} icon-{{nameLowerCase}}" href="#" data-action="{{name}}">' +
```
Comment (PVince81) [inline apps/files/js/fileactions.js]: Why not use "class" from the action object ?

File: apps/files/js/fileactions.js [hunk 2]
```diff
@@ -142,6 +142,7 @@
 				displayName: action.displayName,
 				mime: mime,
 				order: action.order || 0,
+				class: action.class,
```
Comment (PVince81) [inline apps/files/js/fileactions.js]: use this instead of `nameLowerCase` in the template for the icon

Comment (PVince81) [inline apps/files/js/fileactions.js]: and here you could make the class name default to `"icon-" + action.name.toLowerCase()` if not set.
Maybe call it "iconClass" instead ?

File: apps/files/js/filelist.js [hunk 3]
```diff
@@ -2722,7 +2722,7 @@
 			}
 			var $newButton = $(this._addButtonTemplate({
 				addText: t('files', 'New'),
-				iconUrl: OC.imagePath('core', 'actions/add')
+    			class: 'icon-add',
```
Comment (PVince81) [inline apps/files/js/filelist.js]: indent fail

**Comment by jancborchardt** [comment]
The example theme should only have the absolute necessary examples though to be as simple as possible. Action icons aren't part of that – and also the icons.css file should not be copied whole.

Is should be possible by simply adding different action items with the same name in the theme folder. If necessary, adding a CSS rule for those icons. But not duplicating the whole file.

**Comment by rullzer** [comment]
@jancborchardt I wont merge it with the example stuff as is... but I aded them for developing so that we all see the same stuff.

**Comment by rullzer** [comment]
@jancborchardt I do need your help here with the css stuff...

**Comment by PVince81** [comment]
@jancborchardt can you help @rullzer ?

**Comment by PVince81** [comment]
@rullzer it seems there is a problem with the logic.
The actions that have only "iconClass" but no "icon" are rendered as "no-icon" in the menu.
Please change it as follows:
- if neither "icon" not "iconClass" is specified, then render as "no-icon"
- make "iconClass" be empty if none specified, remove the default "icon-" + name logic.

**Comment by PVince81** [comment]
Ok, let me fix this myself, there's additional stuff needed.

**Comment by PVince81** [comment]
Adjusted and squashed.
The file actions icons, the "add" menu, the share icon replacing logic, have been adjusted to use "icon-X" CSS classes instead of "img" elements.

@rullzer I removed the example theme modifications, was confused.

I tried again with a local custom theme, and there is a problem with theming: for some reason, it refuses the load images from "themes/example/core/img/actions" (which I copied from core/img/actions).
I get a 403 Forbidden there. However loading images from "themes/example/core/img/"  works.

@LukasReschke is there a security restriction in place ?

**Comment by PVince81** [comment]
Never mind, my "img/actions" folder had the wrong permissions...

**Comment by PVince81** [comment]
@rullzer @nickvergessen @icewind1991 @schiesbn @jancborchardt @MorrisJobke please review.

To test:

1. `cp core/css/icons.css themes/example/core/css/`
1. `cp -Rv core/img/actions themes/example/core/img`
1. Check the permissions on the "themes/example/core/img/actions" (just in case, happened to me)
1. Maybe swap some icon images, for example `cp star.svg download.svg`
1. Open ownCloud
1. Open the file actions dropdown

You should see a star instead of the download icon.

**Comment by PVince81** [comment]
Also, some regression test needed for all share icon states:
- outgoing link share
- outgoing user share
- incoming share
- switching share state properly updates the share icon
- refresh the page, icons still there

I did test these and they worked.

**Comment by rullzer** [comment]
Fixed the details icon.

**Comment by rullzer** [comment]
- [ ] Bug: I can't see the 3 dots in the public link page (when sharing folders).

**Comment by rullzer** [comment]
- [ ] Missing: Delete icon next to recipients of share is not themed yet

**Comment by PVince81** [comment]
I thought this was only about file actions...

**Comment by rullzer** [comment]
I'm more than fine to leave the missing stuff for a future PR... this is already solving a lot now. Thanks for the fixes @PVince81 .

However the bug on the public link page we should fix

**Comment by PVince81** [comment]
Pfff... now looks like the icons don't have the correct size. The star icon is actually smaller than 16px but the CSS is making it look bigger. Will need more work...

**Comment by PVince81** [comment]
Or... it's the SVG rendering that looks different, weiiiird:

Star before: (img tag)
star-before

Star after: (icon CSS)
star-after

**Comment by PVince81** [comment]
Ok, I had to add `background-size: 16px 16px`.

Please recheck @rullzer

**Comment by PVince81** [comment]
Will fix the JS unit test in a minute...

**Comment by PVince81** [comment]
Done, please review!

**Comment by rullzer** [comment]
:rocket: :+1: all looking good!

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by MorrisJobke** [comment]
Rebased to re-trigger CI

**Comment by jancborchardt** [comment]
Yep, looking good now!

**Pull request merged**